lib/tests/unit/modules/task_test.cc

```diff
@@ -305,7 +306,7 @@ TEST_CASE("Modules::Task::executeAction", "[modules][output]") {
         REQUIRE(boost::contains(response.action_metadata.get<std::string>("execution_error"), "Cannot download task. No master-uris were provided"));
     }
 
-    SECTION("errors when a download error occurs and removes the temporary file") {
+    SECTION("if a master-uri has a server-side error, then it proceeds to try the next master-uri. if they all fail, it errors on download and removes all temporary files") {
```

Comment:
We should also cover the previous error case.

Reply:
I'll add it post-hoc.

Reply:
I guess this test pretty much covers it, just slightly change in the error message.